File: articles.jdbc/src/test/java/pet/db/jdbc/tool/db/DbCleanerImpl.java
package pet.db.jdbc.tool.db;

import lombok.RequiredArgsConstructor;

import org.springframework.stereotype.Component;

import javax.sql.DataSource;
import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.SQLException;

@Component
@RequiredArgsConstructor
public class DbCleanerImpl implements DbCleaner {

    private static final String DELETE_ALL = "DELETE FROM %s";

    private final DataSource dataSource;

    @Override
    public void cleanAll() {
        try {
            cleanTable("reviews");
            cleanTable("authorship_of_articles");
            cleanTable("articles");
            cleanTable("users");
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }}

    private void cleanTable(String tableName) throws SQLException {
        try (Connection connection = dataSource.getConnection();
             PreparedStatement prepareStatement = connection.prepareStatement(
                     String.format(DELETE_ALL, tableName))) {
            prepareStatement.executeUpdate();
        }
    }

}

File: src/test/kotlin/pet/articles/tool/db/DBCleanerImpl.kt
package pet.articles.tool.db


import org.koin.core.annotation.Single
import java.sql.PreparedStatement
import javax.sql.DataSource

class DBCleanerImpl(
    private val dataSource: DataSource
) : DBCleaner {

    companion object {
        private const val DELETE_ALL = "DELETE FROM %s"

        private const val REVIEWS_TABLE_NAME = "public.reviews"
        private const val AUTHORSHIP_TABLE_NAME = "public.authorship_of_articles"
        private const val ARTICLES_TABLE_NAME = "public.articles"
        private const val USERS_TABLE_NAME = "public.users"
    }

    override fun cleanUp() {
        cleanTable(REVIEWS_TABLE_NAME)
        cleanTable(AUTHORSHIP_TABLE_NAME)
        cleanTable(ARTICLES_TABLE_NAME)
        cleanTable(USERS_TABLE_NAME)
    }

    private fun cleanTable(tableName: String) {
        dataSource.connection.use { connection ->
            connection
                .prepareStatement(DELETE_ALL.format(tableName))
                .use(PreparedStatement::executeUpdate)
        }
    }
}